internal/client/client.go

```diff
@@ -175,6 +177,33 @@ func (c *CSAPI) SyncUntil(t *testing.T, since, key string, check func(gjson.Resu
 	}
 }
 
+//RegisterUser will register the user with given parameters and
+// return user ID & access token, and fail the test on network error
+func (c *CSAPI) RegisterUser(t *testing.T, localpart, password string) (userID, accessToken string) {
+	t.Helper()
+	reqBody := map[string]interface{}{
+		"auth": map[string]string{
+			"type": "m.login.dummy",
+		},
+		"username": localpart,
+		"password": password,
+	}
+	res := c.MustDo(t, "POST", []string{"_matrix", "client", "r0", "register"}, reqBody)
+	must.MatchResponse(t, res, match.HTTPResponse{
+		JSON: []match.JSON{
+			match.JSONKeyTypeEqual("access_token", gjson.String),
+			match.JSONKeyTypeEqual("user_id", gjson.String),
+		},
+	})
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatalf("unable to read response body: %v", err)
+	}
+	userID = gjson.Get(string(body), "user_id").String()
```

Comment:
Use `gjson.GetBytes(body, "user_id").Str` instead. `GetBytes` saves a typecast to `string()` and `.Str` won't coerce the value to be a string if it isn't one (e.g if it was a boolean value, `String()=="false"` but `.Str == ""`.